### DISCUSSION TITLE
在vscode中使用debug时，程序不会在断点处暂停

### DISCUSSION THREAD

**Discussion by Konvt** [xmake-io/xmake-vscode#201] [Q&A] [answered]
我在vscode上使用xmake创建了一个c++工程项目，当我尝试在文件中打上断点，用debug模式build文件后，再点击调试按钮时，xmake总是会直接运行到程序结束并正常退出，整个过程都无视了我的断点。

编译器使用mingw，平台windows 10，工程文件路径中没有中文字符。

xmake的debug调试器我分别使用了codelldb选项和default选项，也尝试使用codelldb插件进行调试，但是都会出现断点不暂停的情况。

而且如果开启了入口暂停，被打上的断点还会变成空心点。

下面是程序进入调试时的调试输出。
![VeryCapture_20230602102656](https://github.com/xmake-io/xmake/assets/120050092/4f41f097-c894-4ca8-b41f-8708e56e5c17)

我也尝试使用过cmake工具和vsc自己的调试器进行debug，但只有xmake会有这样的情况。

**Comment by waruqi** [comment]
看下有没有配置 add_rules("mode.debug") 并切到 debug 模式编译。

另外，mingw 不支持生成 pdb ，所以调试只能走 gdb

**Reply by Konvt** [reply]
这里使用的是工程创建时默认生成的 main 文件，xmake.lua 里的 add_rules 确实是带有 mode.debug，且我也切换到了 debug 模式
![VeryCapture_20230602132559](https://github.com/xmake-io/xmake-vscode/assets/120050092/0b827d7e-a1e6-4918-a863-8f39b78bbc3a)

**Comment by Konvt** [comment, marked as answer]
问题已解决

编译时目标平台要选择 mingw 而不是 windows...
